lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResult.cs

```diff
@@ -96,6 +96,36 @@ public string GetUrlResult(string url)
             return task.Result;
         }
 
+        public void WaitForDriverToFinish()
+        {
+            DriverStatus status = FetchDriverStatus();
+            
+            while (status.IsActive())
+            {
+                try
+                {
+                    status = FetchDriverStatus();
+                }
+                catch (System.Net.WebException)
+                {
+                    // If we no longer can reach the Driver, it must have exited.
+                    status = DriverStatus.UNKNOWN_EXITED;
+                }
+            }
+        }
+
+        private DriverStatus FetchDriverStatus()
+        {
+            string statusUrl = DriverUrl + "driverstatus/v1";
```

Comment:
You must make sure the driverurl is available at the time this method is called. Or send web request only when driverUrl is available otherwise return status like UNKOWN or INIT.

Reply:
Doesn't the class make sure of that as is? `DriverUrl` merely refers to `_driverUrl`, which is set in the constructor by calling `GetDriverUrl`, which is to be implemented by every subclass. I'm assuming the subclasses to work.

If they don't, then this call will throw a `WebException`, which is documented behavior and also called out in the `Unstable` attribute. 

What else does this code need to check?

Reply:
May be we should add sleep/retry logic in GetTrackingUrlAppId method if the file does not exist or driverUrl is not available yet. Another improvement could be to modify DriverUrl property to check if _driverUrl contains proper url, if not, recall GetDriverUrl again. It can be in separate JIRA though. 
I have created REEF-1946 for the follow up.

Reply:
Sounds good to treat this separately. Long term, I don't like this whole polling business and would much rather see a event driven approach. Let's discuss that as part of [REEF-1946]